griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,436 @@
+        from googleapiclient.errors import HttpError
+
+        values = params["values"]
+        file_paths = values.get("file_paths")
+        mime_type = values.get("mime_type")
+
+        export_mime_mapping = {
+            "text/csv": "text/csv",
+            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        }
+
+        downloaded_files = []
+
+        if mime_type not in export_mime_mapping:
+            logging.error(f"Unsupported mime type '{mime_type}' provided.")
+            return ErrorArtifact("Unsupported mime type for downloading")
+        else:
+            try:
+                service = self._build_client(
+                    scopes=self.DRIVE_READ_SCOPES,
+                    service_name="drive",
+                    version="v3",
+                    owner_email=self.owner_email,
+                )
+                for file_path in file_paths:
+                    sheet_id = self._convert_path_to_file_id(service, file_path)
+
+                    if sheet_id:
+                        request = service.files().export_media(
+                            fileId=sheet_id,
+                            mimeType=export_mime_mapping[mime_type],
+                        )
+                        downloaded_files.append(BlobArtifact(request.execute()))
+                    else:
+                        logging.error(
+                            f"error finding spreadsheet at path: {file_path}"
+                        )
+
+            except HttpError as e:
+                return ErrorArtifact(
+                    f"error downloading spreadsheet due to http error: {e}"
+                )
+            except MalformedError as e:
+                return ErrorArtifact(
+                    f"error downloading spreadsheet due to malformed credentials: {e}"
+                )
+            except Exception as e:
+                return ErrorArtifact(f"error downloading spreadsheets: {e}")
+
+        return ListArtifact(downloaded_files)
+
+    @activity(
+        config={
+            "description": "Uploads a spreadsheet and converts it to a Google Sheets format",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_name",
+                        description="The name of the file to be uploaded",
+                    ): str,
+                    Literal(
+                        "file_path",
+                        description="The local path to the file to be uploaded",
+                    ): str,
+                    Literal(
+                        "file_type",
+                        description="The type of the file being uploaded, e.g., 'csv' or 'excel'",
+                    ): str,
+                }
+            ),
+        }
+    )
+    def upload_spreadsheet(self, params: dict) -> InfoArtifact | ErrorArtifact:
```

Comment:
I think this should live in the Drive tool as `upload_file`.

Reply:
Check Slack!